src/components/login/AuthDetails.tsx

```diff
@@ -0,0 +1,28 @@
+import { onAuthStateChanged, signOut } from "firebase/auth";
+import { useEffect, useState } from "react";
+import { auth } from '../../firebase-config';
+
+const AuthDetails = () => {
+    const [authUser, setAuthUser] = useState(null);
```

Comment:
Make sure to specify the type User or null for your hooks.
Like this:
`  const [authUser, setAuthUser] = useState<User | null>(null);`

Reply:
Done. I'm still getting familiar and functionality of hooks and useState